relayer/src/chain/cosmos.rs

```diff
@@ -266,10 +307,9 @@ impl Chain for CosmosSDKChain {
 
         let mut txraw_buf = Vec::new();
         prost::Message::encode(&tx_raw, &mut txraw_buf).unwrap();
-        //println!("TxRAW {:?}", hex::encode(txraw_buf.clone()));
 
-        //let signed = sign(sign_doc);
-        let response = block_on(broadcast_tx(self, txraw_buf)).map_err(|e| Kind::Rpc.context(e))?;
+        let response =
+            block_on(broadcast_tx_sync(self, txraw_buf)).map_err(|e| Kind::Rpc.context(e))?;
```

Comment:
It was a bit of surprise not to see `broadcast_tx_commit`. Do we proceed with `_sync`?

Reply:
There is a bug in tendermint-rpc wrt to decoding the response. It will always return error. The advantage of using `broadcast_tx_commit` is that it will show the `DeliverTx` errors. With `broadcast_tx_sync` you need to dig out from the gaia logs. I have switched to `broadcast_tx_commit` in my latest commit as I was doing more tests. I am ok with using `broadcast_tx_commit` in master also but again, all CLIs will show some errors even though the Tx-es are successful.

Reply:
tm rpc issue: https://github.com/informalsystems/tendermint-rs/issues/667